source/autocomplete/autocomplete.h

```diff
@@ -493,18 +471,18 @@ struct Autocomplete
         strFind = strip_accents(strFind);
         std::string strTemp = strFind;
 
-        //if synonyms contains something, add all synonyms if found while serching on ket et value.
+        //if synonyms contains something, add all synonyms if found while serching on key and value.
         //For each synonyms.key found in strFind add synonyms.value
         for(const auto& it : synonyms){
-            if  (boost::regex_search(strFind,boost::regex("\\<" + it.first + "\\>"))){
-                strTemp += " " + it.second;
+            if  (boost::regex_search(strFind,boost::regex("\\<" + strip_accents_and_lower(it.first) + "\\>"))){
+                strTemp += " " + strip_accents_and_lower(it.second);
             }
         }
 
         //For each synonyms.value found in strFind add synonyms.key
         for(const auto& it : synonyms){
-            if  (boost::regex_search(strFind,boost::regex("\\<" + it.second + "\\>"))){
-                strTemp += " " + it.first;
+            if  (boost::regex_search(strFind,boost::regex("\\<" + strip_accents_and_lower(it.second) + "\\>"))){
+                strTemp += " " + strip_accents_and_lower(it.first);
             }
         }
 
```

Comment:
The use of  function "strip_accents_and_lower" in the tokenizer is timeconsuming. It would be preferable if we can apply the function  for all the synonyms during binarization. Apart this it is fine for me.